lib/hippie/client.js

```diff
@@ -13,6 +13,7 @@ var http = require('http');
 var serializers = require('./serializers');
 var parsers = require('./parsers');
 var expect = require('./expect');
+var Promise = require('es6-promise').Promise;
```

Comment:
Hmmm right...  I see the point, though I might need some help with the best way to implement it.

Looking around, it seems like most libraries either 
1) Pick their preferred promise library to use internally (ref: Sequelize, Mongoose), i.e. the approach of this PR.
2) Allow you to pass in an options hash w/the promise library of your choice by invoking the require, e.g.
```js
var options = {
    // promiseLib - overrides default promise library;
};
var pgp = require('pg-promise')(options);
```

My reservation with 2) is that it would drastically change how hippie is invoked.  i.e. rather than ```require('hippie')```, it would now be ```require('hippie')()```.  Upgrade path would suck.

3)  There's the approach of adding a function onto the hippie constructor itself, which would apply the override across all new instances of hippie:
```js
Client.options = function(options) {
  //override default promise lib
};
```
which yields:
```js
   hippie = require('hippie') //same as before
   hippie.options({promiseLib: require('bluebird').Promise})

   hippie(app)
   ...

   hippie(app)
   ...

   hippie(app)
   ...
```

Any other option you can see that I'm missing?

Personally, I think as time goes by and more people upgrade newer version of node w/native promises, usage of custom promise libraries will quickly recede.  That said, I'm happy to take whichever approach you prefer!